### ISSUE TITLE
valid-jsdoc: optional parameter type

### ISSUE THREAD

**Issue by Conaclos** [eslint/eslint#8988]
**Tell us about your environment**

* **ESLint Version:** 4.3.0
* **Node Version:** 8.2.1
* **npm Version:** 5.3.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    ...
    "valid-jsdoc": [2, {
        requireReturn: false,
        requireReturnType: false,
        requireParamDescription: false,
        requireReturnDescription: false
    }
}
```
</details>

**What did you do? Please include the actual source code causing the issue.**

```js
/**
 *
 * @param a
 * @return a result
 */
function f (a) {
    return a
}
```

**What did you expect to happen?** The declaration of the argument type should be optional. It could be great to have an option "requireParameterType".

**What actually happened? Please include the actual, raw output from ESLint.**

ESLint complains about a missing type for parameter `a`.

**Comment by not-an-aardvark** [comment]
Thanks for the issue, this seems reasonable to me.

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
